### PR TITLE
[Docs] add sign up button to nav bar

### PR DESCRIPTION
**Pull request by ryancurtis1**
Adds a sign up button to the nav bar on docs, only when there is room

Also updates the background color of nav bar in dark mode per design

Example (Light):
Screenshot 2024-10-03 at 6 08 11 PM

Dark Mode:
Screenshot 2024-10-03 at 6 08 39 PM

### CODE REVIEW

File: components/SignUpButton/SignUpButton.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import style from "./SignUpButton.module.scss";
+
+export default function SignUpButton() {
+  return (
+    <div className={style.signUpButton}>
```
Comment (tiffanyqi) [inline components/SignUpButton/SignUpButton.tsx]: can we flip it so that `a` is outside of the div? i think that's more proper

**Review by tiffanyqi — COMMENTED** [review]
do we also want a signup button in the sidenav if we get into mobile?

**Comment by ryancurtis1** [comment]
On small screens:

< 400px:
Screenshot 2024-10-04 at 2 04 48 PM

400-767px:
Screenshot 2024-10-04 at 2 06 08 PM

768-800px:
Screenshot 2024-10-04 at 2 08 58 PM

**Review by myronkaifung — APPROVED** [review]

**Comment by ryancurtis1** [comment]
> do we also want a signup button in the sidenav if we get into mobile?

Synced with Tara, and we will keep the sign up button in the nav bar for small

File: components/SignUpButton/SignUpButton.tsx (hunk 1 above)
Comment (ryancurtis1) [reply components/SignUpButton/SignUpButton.tsx]: 👍

**Pull request merged**